File: LivrusVirtus/FoxToolKit/FXPython/CadEnderecos.py
from FXPy.fox import *
from Cadastros import frmCadastros
from resource import Resource
from RotinasGlobais import RotinasGlobais
from ConsultasSQL import ConsultasSQL
import Principal

class frmCadEnderecos(frmCadastros):

  def __init__(self, clienteMDI, iconeMDI, menuMDI):  
    frmCadastros.__init__(self, clienteMDI, Resource.STR_CADENDERECO, 
      iconeMDI, menuMDI, TEXTFIELD_INTEGER, False)

    self.Rotinas = RotinasGlobais.Rotinas
    self.ConsSQL = ConsultasSQL.ConsSQL

    FXMAPFUNC(self, SEL_COMMAND, frmCadEnderecos.ID_FRMCADASTROS_BTNNOVO, 
      frmCadEnderecos.onBtnNovo)
    FXMAPFUNC(self, SEL_COMMAND, frmCadEnderecos.ID_FRMCADASTROS_BTNPESQUISAR, 
      frmCadEnderecos.onBtnPesquisar)
    FXMAPFUNC(self, SEL_COMMAND, frmCadEnderecos.ID_FRMCADASTROS_BTNANTERIOR, 
      frmCadEnderecos.onBtnAnterior)
    FXMAPFUNC(self, SEL_COMMAND, frmCadEnderecos.ID_FRMCADASTROS_BTNPROXIMO, 
      frmCadEnderecos.onBtnProximo)
    FXMAPFUNC(self, SEL_COMMAND, frmCadEnderecos.ID_FRMCADASTROS_BTNSALVAR, 
      frmCadEnderecos.onBtnSalvar)
    FXMAPFUNC(self, SEL_COMMAND, frmCadEnderecos.ID_FRMCADASTROS_BTNEXCLUIR, 
      frmCadEnderecos.onBtnExcluir)
    FXMAPFUNC(self, SEL_COMMAND, frmCadEnderecos.ID_FRMCADASTROS_BTNLIMPAR, 
      frmCadEnderecos.onBtnLimpar)
    FXMAPFUNC(self, SEL_COMMAND, frmCadEnderecos.ID_FRMCADASTROS_BTNFECHAR, 
      frmCadEnderecos.onBtnFechar)
    FXMAPFUNC(self, SEL_CLOSE, 0, frmCadEnderecos.onClose)

    self.resize(540, 225)
    self.labDescricao.setText("Logradouro: ")
    self.labDescricao.resize(57, 13)
    self.edtDescricao.resize(264, 21)
    self.edtDescricao.setNumColumns(40)
  
    frame9 = FXVerticalFrame(self.frame7, LAYOUT_LEFT, 0,0,0,0,0,0,0,0)
  
    frame10 = FXHorizontalFrame(self.frame3, LAYOUT_LEFT, 0,0,0,0,0,0,0,0)
    frame11 = FXVerticalFrame(frame10, LAYOUT_LEFT, 0,0,0,0,0,0,0,0)
    frame12 = FXVerticalFrame(frame10, LAYOUT_LEFT, 0,0,0,0,0,0,0,0)
    frame13 = FXVerticalFrame(frame10, LAYOUT_LEFT, 0,0,0,0,0,0,0,0)
  
    labBairro = FXLabel(frame9, "Bairro:", None, LAYOUT_LEFT, 
      280, 48, 48, 13)
    self.edtBairro = FXTextField(frame9, 15, None, 0, TEXTFIELD_LIMITED|
      FRAME_SUNKEN|FRAME_THICK|LAYOUT_FIX_WIDTH|LAYOUT_FIX_HEIGHT, 
      280, 64, 144, 21, 3, 20)
    self.edtBairro.disable()
    labCEP = FXLabel(frame11, "CEP:", None, LAYOUT_LEFT, 
      8, 88, 40, 13)
    self.edtCEP = FXTextField(frame11, 8, None, 0, TEXTFIELD_LIMITED|
      FRAME_SUNKEN|FRAME_THICK|LAYOUT_FIX_WIDTH|LAYOUT_FIX_HEIGHT, 
      8, 104, 72, 21, 3, 20)
    self.edtCEP.disable()
    labCidade = FXLabel(frame12, "Cidade:", None, LAYOUT_LEFT, 
      88, 88, 40, 13)
    self.edtCidade = FXTextField(frame12, 30, None, 0, TEXTFIELD_LIMITED|
      FRAME_SUNKEN|FRAME_THICK|LAYOUT_FIX_WIDTH|LAYOUT_FIX_HEIGHT, 
      88, 104, 184, 21, 3, 20)
    self.edtCidade.disable()
    labEstado = FXLabel(frame13, "Estado:", None, LAYOUT_LEFT, 
      280, 88, 56, 13)
    self.cmbEstado = FXComboBox(frame13, 8, 8, None, 0,
      COMBOBOX_INSERT_LAST|FRAME_SUNKEN|FRAME_THICK|LAYOUT_SIDE_TOP|
      LAYOUT_FIX_WIDTH|LAYOUT_FIX_HEIGHT, 280, 104, 144, 21, 3, 20)
    self.cmbEstado.setEditable(False)
    self.cmbEstado.setNumVisible(8)
    self.cmbEstado.disable()
  
    self.AdicionaCombo()
    self.IniciaBotoes()

  def ValidaDados(self, bTodosDados):
    if self.Rotinas.ValidaCampo(self.edtCodigo.getText(), Resource.STR_CODNINF):
   	  return False   
    if bTodosDados:
  	  if self.Rotinas.ValidaCampo(self.edtDescricao.getText(), Resource.STR_ENDNINF):
    		return False
  	  if self.Rotinas.ValidaCampo(self.edtBairro.getText(), Resource.STR_BAIRRONINF):
    		return False
  	  if self.Rotinas.ValidaCampo(self.edtCEP.getText(), Resource.STR_CEPNINF):
    		return False
  	  if self.Rotinas.ValidaCampo(self.edtCidade.getText(), Resource.STR_CIDADENINF):
    		return False
    return True
  
  def onBtnNovo(self, sender, sel, ptr):
    self.btnNovoClick()
    self.InformaLimpaDadosLocal(False)
    self.HabilitaDadosLocal(True)	
    self.NovoDado(self.ConsSQL.Endereco('N', self.edtCodigo.getText(),"","","","",""))	
    return True
    
  def onBtnPesquisar(self, sender, sel, ptr):
    if self.PesquisarDados(self.ConsSQL.Endereco('S', self.edtCodigo.getText(),
      "","","","",""), Resource.STR_ENDNENC):
      self.InformaLimpaDadosLocal(True)
      self.HabilitaDadosLocal(True)    
    return True  
  
  def onBtnAnterior(self, sender, sel, ptr):
    if self.registroAnterior():
      self.InformaLimpaDadosLocal(True)    
    return True
    
  def onBtnProximo(self, sender, sel, ptr):
    if self.registroProximo():
      self.InformaLimpaDadosLocal(True)    
    return True  
  
  def onBtnSalvar(self, sender, sel, ptr):
    if self.SalvarDados(self.ValidaDados(True),
      self.ConsSQL.Endereco('S', self.edtCodigo.getText(),"","","","",""),
      self.ConsSQL.Endereco('U', self.edtCodigo.getText(), self.edtDescricao.getText(),
        self.edtBairro.getText(), self.edtCEP.getText(), self.edtCidade.getText(), 
        self.cmbEstado.getText()),
      self.ConsSQL.Endereco('I', self.edtCodigo.getText(), self.edtDescricao.getText(),
        self.edtBairro.getText(), self.edtCEP.getText(), self.edtCidade.getText(), 
        self.cmbEstado.getText()), True):
      self.InformaLimpaDadosLocal(False)
      self.HabilitaDadosLocal(False)    
    return True  
  
  def onBtnExcluir(self, sender, sel, ptr):
    if self.ExcluirDados(ValidaDados(False),
      self.ConsSQL.Endereco('D', self.edtCodigo.getText(),"","","","","")):
      self.InformaLimpaDadosLocal(False)
      self.HabilitaDadosLocal(False)    
    return True  
  
  def onBtnLimpar(self, sender, sel, ptr):
    self.btnLimparClick()
    self.InformaLimpaDadosLocal(False)
    self.HabilitaDadosLocal(False)
    return True  

  def onBtnFechar(self, sender, sel, ptr):
    self.onClose(sender, sel, ptr)
    return True  

  def onClose(self, sender, sel, ptr):
    Principal.fPrincipal.habilitaMenu("CadEnderecos")
    self.destroy()
    return True   
 
  def InformaLimpaDadosLocal(self, bInformar):
    if bInformar:
      self.edtBairro.setText(self.lstRegistros.col[self.nLinhaRegistro][2])
      self.edtCEP.setText(self.lstRegistros.col[self.nLinhaRegistro][3])
      self.edtCidade.setText(self.lstRegistros.col[self.nLinhaRegistro][4])
      self.cmbEstado.setText(self.lstRegistros.col[self.nLinhaRegistro][5])
    else:
      self.edtBairro.setText("")
      self.edtCEP.setText("")
      self.edtCidade.setText("")
      self.cmbEstado.setCurrentItem(0)         
  
  def HabilitaDadosLocal(self, bHabilita):
    if bHabilita:
      self.edtBairro.enable()
      self.edtCEP.enable()
      self.edtCidade.enable()
      self.cmbEstado.enable()
    else:
      self.edtBairro.disable()
      self.edtCEP.disable()
      self.edtCidade.disable()
      self.cmbEstado.disable()      
  
  def AdicionaCombo(self):
    self.cmbEstado.appendItem(Resource.STR_PE)
    self.cmbEstado.appendItem(Resource.STR_AC)
    self.cmbEstado.appendItem(Resource.STR_AL)
    self.cmbEstado.appendItem(Resource.STR_AP)
    self.cmbEstado.appendItem(Resource.STR_AM)
    self.cmbEstado.appendItem(Resource.STR_BA)
    self.cmbEstado.appendItem(Resource.STR_BR)
    self.cmbEstado.appendItem(Resource.STR_CE)
    self.cmbEstado.appendItem(Resource.STR_DF)
    self.cmbEstado.appendItem(Resource.STR_ES)
    self.cmbEstado.appendItem(Resource.STR_GO)
    self.cmbEstado.appendItem(Resource.STR_MT)
    self.cmbEstado.appendItem(Resource.STR_MS)
    self.cmbEstado.appendItem(Resource.STR_MG)
    self.cmbEstado.appendItem(Resource.STR_PA)
    self.cmbEstado.appendItem(Resource.STR_PB)
    self.cmbEstado.appendItem(Resource.STR_PR)
    self.cmbEstado.appendItem(Resource.STR_SC)
    self.cmbEstado.appendItem(Resource.STR_RN)
    self.cmbEstado.appendItem(Resource.STR_RS)
    self.cmbEstado.appendItem(Resource.STR_RJ)
    self.cmbEstado.appendItem(Resource.STR_RO)
    self.cmbEstado.appendItem(Resource.STR_RR)
    self.cmbEstado.appendItem(Resource.STR_SP)
    self.cmbEstado.appendItem(Resource.STR_SE)
    self.cmbEstado.appendItem(Resource.STR_TO)

File: LivrusVirtus/FoxToolKit/FXRuby/CadEnderecos.rb
require 'fox'
require 'fox/responder'
require 'Cadastros'
require 'Principal'
require 'RotinasGlobais'
require 'ConsultasSQL'
include Fox
include Responder

class FRMCadEnderecos < FRMCadastros
  
  def initialize(clienteMDI, iconeMDI, menuMDI)  
    super(clienteMDI, Resource::STR_CADENDERECO, iconeMDI, menuMDI, 
      TEXTFIELD_INTEGER, false)

    FXMAPFUNC(SEL_COMMAND, ID_FRMCADASTROS_BTNNOVO, :onBtnNovo)
    FXMAPFUNC(SEL_COMMAND, ID_FRMCADASTROS_BTNPESQUISAR, :onBtnPesquisar)
    FXMAPFUNC(SEL_COMMAND, ID_FRMCADASTROS_BTNANTERIOR, :onBtnAnterior)
    FXMAPFUNC(SEL_COMMAND, ID_FRMCADASTROS_BTNPROXIMO, :onBtnProximo)
    FXMAPFUNC(SEL_COMMAND, ID_FRMCADASTROS_BTNSALVAR, :onBtnSalvar)
    FXMAPFUNC(SEL_COMMAND, ID_FRMCADASTROS_BTNEXCLUIR, :onBtnExcluir)
    FXMAPFUNC(SEL_COMMAND, ID_FRMCADASTROS_BTNLIMPAR, :onBtnLimpar)
    FXMAPFUNC(SEL_COMMAND, ID_FRMCADASTROS_BTNFECHAR, :onBtnFechar)
    FXMAPFUNC(SEL_CLOSE, 0, :onClose)

    resize(540, 225)
    @labDescricao.setText("Logradouro: ")
    @labDescricao.resize(57, 13)
    @edtDescricao.resize(264, 21)
    @edtDescricao.setNumColumns(40)
  
    frame9 = FXVerticalFrame.new(@frame7, LAYOUT_LEFT, 0,0,0,0,0,0,0,0)
  
    frame10 = FXHorizontalFrame.new(@frame3, LAYOUT_LEFT, 0,0,0,0,0,0,0,0)
    frame11 = FXVerticalFrame.new(frame10, LAYOUT_LEFT, 0,0,0,0,0,0,0,0)
    frame12 = FXVerticalFrame.new(frame10, LAYOUT_LEFT, 0,0,0,0,0,0,0,0)
    frame13 = FXVerticalFrame.new(frame10, LAYOUT_LEFT, 0,0,0,0,0,0,0,0)
  
    labBairro = FXLabel.new(frame9, "Bairro:", nil, LAYOUT_LEFT, 
      280, 48, 48, 13)
    @edtBairro = FXTextField.new(frame9, 15, nil, 0, TEXTFIELD_LIMITED|
      FRAME_SUNKEN|FRAME_THICK|LAYOUT_FIX_WIDTH|LAYOUT_FIX_HEIGHT, 
      280, 64, 144, 21, 3, 20)
    @edtBairro.disable()
    labCEP = FXLabel.new(frame11, "CEP:", nil, LAYOUT_LEFT, 
      8, 88, 40, 13)
    @edtCEP = FXTextField.new(frame11, 8, nil, 0, TEXTFIELD_LIMITED|
      FRAME_SUNKEN|FRAME_THICK|LAYOUT_FIX_WIDTH|LAYOUT_FIX_HEIGHT, 
      8, 104, 72, 21, 3, 20)
    @edtCEP.disable()
    labCidade = FXLabel.new(frame12, "Cidade:", nil, LAYOUT_LEFT, 
      88, 88, 40, 13)
    @edtCidade = FXTextField.new(frame12, 30, nil, 0, TEXTFIELD_LIMITED|
      FRAME_SUNKEN|FRAME_THICK|LAYOUT_FIX_WIDTH|LAYOUT_FIX_HEIGHT, 
      88, 104, 184, 21, 3, 20)
    @edtCidade.disable()
    labEstado = FXLabel.new(frame13, "Estado:", nil, LAYOUT_LEFT, 
      280, 88, 56, 13)
    @cmbEstado = FXComboBox.new(frame13, 8, 8, nil, 0,
      COMBOBOX_INSERT_LAST|FRAME_SUNKEN|FRAME_THICK|LAYOUT_SIDE_TOP|
      LAYOUT_FIX_WIDTH|LAYOUT_FIX_HEIGHT, 280, 104, 144, 21, 3, 20)
    @cmbEstado.setEditable(false)
    @cmbEstado.setNumVisible(8)
    @cmbEstado.disable()
  
    AdicionaCombo()
    IniciaBotoes()
  end

  def ValidaDados(bTodosDados)
    if $Rotinas.ValidaCampo(@edtCodigo.getText(), Resource::STR_CODNINF)
   	  return false
    end
  	if bTodosDados
  	  if $Rotinas.ValidaCampo(@edtDescricao.getText(), Resource::STR_ENDNINF)
    		return false
      end
  	  if $Rotinas.ValidaCampo(@edtBairro.getText(), Resource::STR_BAIRRONINF)
    		return false
      end
  	  if $Rotinas.ValidaCampo(@edtCEP.getText(), Resource::STR_CEPNINF)
    		return false
      end
  	  if $Rotinas.ValidaCampo(@edtCidade.getText(), Resource::STR_CIDADENINF)
    		return false
      end
  	end
  	return true
  end 
  
  def onBtnNovo(sender, sel, ptr)
    btnNovoClick()
    InformaLimpaDadosLocal(false)
    HabilitaDadosLocal(true)	
    NovoDado($ConsSQL.Endereco('N', @edtCodigo.getText(),"","","","",""))	
    return true
  end
  
  def onBtnPesquisar(sender, sel, ptr)
    if PesquisarDados($ConsSQL.Endereco('S', @edtCodigo.getText(),
      "","","","",""), Resource::STR_ENDNENC)
      InformaLimpaDadosLocal(true)
      HabilitaDadosLocal(true)
    end
    return true
  end
  
  def onBtnAnterior(sender, sel, ptr)
    if registroAnterior()	
      InformaLimpaDadosLocal(true)
    end
    return true
  end
  
  def onBtnProximo(sender, sel, ptr)
    if registroProximo()
      InformaLimpaDadosLocal(true)
    end
    return true
  end
  
  def onBtnSalvar(sender, sel, ptr)
    if SalvarDados(ValidaDados(true),
      $ConsSQL.Endereco('S', @edtCodigo.getText(),"","","","",""),
      $ConsSQL.Endereco('U', @edtCodigo.getText(), @edtDescricao.getText(),
        @edtBairro.getText(), @edtCEP.getText(), @edtCidade.getText(), 
        @cmbEstado.getText()),
      $ConsSQL.Endereco('I', @edtCodigo.getText(), @edtDescricao.getText(),
        @edtBairro.getText(), @edtCEP.getText(), @edtCidade.getText(), 
        @cmbEstado.getText()), true)
      InformaLimpaDadosLocal(false)
      HabilitaDadosLocal(false)
    end
    return true
  end
  
  def onBtnExcluir(sender, sel, ptr)
    if ExcluirDados(ValidaDados(false),
      $ConsSQL.Endereco('D', @edtCodigo.getText(),"","","","",""))
      InformaLimpaDadosLocal(false)
      HabilitaDadosLocal(false)
    end
    return true
  end
  
  def onBtnLimpar(sender, sel, ptr)
    btnLimparClick()
    InformaLimpaDadosLocal(false)
    HabilitaDadosLocal(false)
    return true
  end

  def onBtnFechar(sender, sel, ptr)
    onClose(sender, sel, ptr)
    return true
  end

  def onClose(sender, sel, ptr)
    $frmPrincipal.habilitaMenu("CadEnderecos")
    destroy()
    return true
  end 
 
  def InformaLimpaDadosLocal(bInformar)
    if bInformar
      @edtBairro.setText(@lstRegistros.col[@nLinhaRegistro][2])
      @edtCEP.setText(@lstRegistros.col[@nLinhaRegistro][3])
      @edtCidade.setText(@lstRegistros.col[@nLinhaRegistro][4])
      @cmbEstado.setText(@lstRegistros.col[@nLinhaRegistro][5])
    else
      @edtBairro.setText("")
      @edtCEP.setText("")
      @edtCidade.setText("")
      @cmbEstado.setCurrentItem(0)
    end  
  end 
  
  def HabilitaDadosLocal(bHabilita)
    if bHabilita
      @edtBairro.enable()
      @edtCEP.enable()
      @edtCidade.enable()
      @cmbEstado.enable()
    else
      @edtBairro.disable()
      @edtCEP.disable()
      @edtCidade.disable()
      @cmbEstado.disable()
    end
  end
  
  def AdicionaCombo()
    @cmbEstado.appendItem(Resource::STR_PE)
    @cmbEstado.appendItem(Resource::STR_AC)
    @cmbEstado.appendItem(Resource::STR_AL)
    @cmbEstado.appendItem(Resource::STR_AP)
    @cmbEstado.appendItem(Resource::STR_AM)
    @cmbEstado.appendItem(Resource::STR_BA)
    @cmbEstado.appendItem(Resource::STR_BR)
    @cmbEstado.appendItem(Resource::STR_CE)
    @cmbEstado.appendItem(Resource::STR_DF)
    @cmbEstado.appendItem(Resource::STR_ES)
    @cmbEstado.appendItem(Resource::STR_GO)
    @cmbEstado.appendItem(Resource::STR_MT)
    @cmbEstado.appendItem(Resource::STR_MS)
    @cmbEstado.appendItem(Resource::STR_MG)
    @cmbEstado.appendItem(Resource::STR_PA)
    @cmbEstado.appendItem(Resource::STR_PB)
    @cmbEstado.appendItem(Resource::STR_PR)
    @cmbEstado.appendItem(Resource::STR_SC)
    @cmbEstado.appendItem(Resource::STR_RN)
    @cmbEstado.appendItem(Resource::STR_RS)
    @cmbEstado.appendItem(Resource::STR_RJ)
    @cmbEstado.appendItem(Resource::STR_RO)
    @cmbEstado.appendItem(Resource::STR_RR)
    @cmbEstado.appendItem(Resource::STR_SP)
    @cmbEstado.appendItem(Resource::STR_SE)
    @cmbEstado.appendItem(Resource::STR_TO)
  end

end
